crosshair/diff_behavior.py

```diff
@@ -228,9 +268,19 @@ def run_iteration(
         result1 = describe_behavior(fn1, args1)
         result2 = describe_behavior(fn2, args2)
         space.detach_path()
-        if flexible_equal(result1, result2) and flexible_equal(args1, args2):
-            debug("Functions equivalent")
-            return (VerificationStatus.CONFIRMED, None)
+        if flexible_equal(args1, args2) and (
+            flexible_equal(result1[0], result2[0])  # Compare only the results.
+        ):
```

Comment:
Swap the order of these clauses to fix that strange `test_diff_behavior_nan` failure. (`flexible_equal(result1[0], result2[0]) and flexible_equal(args1, args2)`)

There's actually a latent bug in the nan comparison thing I just did, but it only gets exposed because you've changed the order of the comparison, and that's enough to make the system pick different paths in the test. I'll fix the root cause separately (and increase the iterations in the test so that we would be able to detect a regression in the future)

Reply:
Thanks!